google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1546,18 +1613,28 @@ def load_job files, format: nil, create: nil, write: nil,
                      skip_leading: nil, dryrun: nil, job_id: nil, prefix: nil,
                      labels: nil, autodetect: nil, null_marker: nil
           ensure_service!
-          options = { format: format, create: create, write: write,
-                      projection_fields: projection_fields,
-                      jagged_rows: jagged_rows,
-                      quoted_newlines: quoted_newlines, encoding: encoding,
-                      delimiter: delimiter, ignore_unknown: ignore_unknown,
-                      max_bad_records: max_bad_records, quote: quote,
-                      skip_leading: skip_leading, dryrun: dryrun,
-                      job_id: job_id, prefix: prefix, labels: labels,
-                      autodetect: autodetect, null_marker: null_marker }
-          return load_storage(files, options) if storage_url? files
-          return load_local(files, options) if local_file? files
-          raise Google::Cloud::Error, "Don't know how to load #{files}"
+
+          updater = load_job_updater format: format, create: create,
+                                     write: write,
+                                     projection_fields: projection_fields,
+                                     jagged_rows: jagged_rows,
+                                     quoted_newlines: quoted_newlines,
+                                     encoding: encoding,
+                                     delimiter: delimiter,
+                                     ignore_unknown: ignore_unknown,
+                                     max_bad_records: max_bad_records,
+                                     quote: quote, skip_leading: skip_leading,
+                                     dryrun: dryrun, schema: schema,
+                                     labels: labels, autodetect: autodetect,
+                                     null_marker: null_marker
+
+          yield updater if block_given?
+
+          job_gapi = updater.to_gapi
+          if local_file? files
```

Comment:
I assume `local_file?` is okay getting a list now?

Reply:
No, only one local file can be provided. An array of URLs can be provided, either as strings or URI objects or storage file objects.

Reply:
Does it return false when given array? That sounds like it should be the correct behavior.

Reply:
@tswast I struggled to understand this as well while doing the rebase.

I don't think the docs make it clear that only one local file can be provided:

> A file or the URI of a Google Cloud Storage file, or an Array of  those, containing data to load into the table.

However, I'm fairly confident that in this case I preserved the original behavior of #2000 in my manual conflict resolution.

Reply:
Yes, `local_file?` will return `false` if passed an array.

Reply:
Maybe a doc update if this is confusing?

Also , why not use `load_local_or_uri` here?